tap_github/tests/__init__.py

```diff
@@ -1 +1,19 @@
 """Test suite for tap-github."""
+import requests
+import requests_cache
+
+# Setup caching for all api calls done through `requests` in order to limit
+# rate limiting problems with github.
+# Use the sqlite backend as it's the default option and seems to be best supported.
+# To clear the cache, just delete the sqlite db file at api_calls_tests_cache.sqlite
+# in the root of this repository
+requests_cache.install_cache(
+    "api_calls_tests_cache",
+    backend="sqlite",
+    # make sure that API keys don't end up being cached
+    ignored_parameters=["Authorization"],
+    # tell requests_cache to check headers for the above parameter
+    match_headers=True,
+    # expire the cache after 24h (86400 seconds)
+    expire_after=24 * 60 * 60,
+)
```

Comment:
@laurentS - This is great; I'm very impressed how little code is needed! Also, I didn't know exactly how "ignored_parameters" would turn out but "Authorization" is a standard header which also seems a (somewhat) portable solution.

Did you scan the SQLite DB contents to see if there were any other auth or sensitive tokens captured?

Would be great to see this added to the SDK's cookiecutter - or perhaps into the `Code Samples` page on the docs.

cc @edgarrmondragon 

PS - Meltano is officially OOO the next two weeks so please pardon any delays in communications during the meanwhile. 😄

Reply:
Nothing that looks problematic, no. Here's an example of the content for one entry taken from the sqlite database (after I changed the serializer from `pickle` to `json` for readability :sweat_smile:). Most importantly, the `Authorization` header is stripped as expected. So I think it's safe to cache this.

As for adding this into the SDK, I think pushing this to code samples would make more sense, as I don't know how generalisable the config is.

```json
{
  "content": "<content of the response, encoded>"
  "created_at": "2021-12-20T15:11:43.947734",
  "elapsed": 0.295303,
  "encoding": "utf-8",
  "expires": "2021-12-21T15:11:43.947679",
  "headers": {
    "Server": "GitHub.com",
    "Date": "Mon, 20 Dec 2021 15:11:43 GMT",
    "Content-Type": "application/json; charset=utf-8",
    "Transfer-Encoding": "chunked",
    "Cache-Control": "private, max-age=60, s-maxage=60",
    "Vary": "Accept, Authorization, Cookie, X-GitHub-OTP, Accept-Encoding, Accept, X-Requested-With",
    "ETag": "W/\"d6a3e7fa8493843d598ab6fecb3775a3eb80e77269baf7425371d0c231b43ab4\"",
    "X-OAuth-Scopes": "",
    "X-Accepted-OAuth-Scopes": "",
    "X-GitHub-Media-Type": "github.v3; param=star; format=json",
    "Link": "<https://api.github.com/repositories/1296269/stargazers?per_page=100&page=12&sort=updated&direction=asc&since=2021-12-20T00%3A00%3A00%2B00%3A00>; rel=\"prev\", <https://api.github.com/repositories/1296269/stargazers?per_page=100&page=14&sort=updated&direction=asc&since=2021-12-20T00%3A00%3A00%2B00%3A00>; rel=\"next\", <https://api.github.com/repositories/1296269/stargazers?per_page=100&page=18&sort=updated&dir
ection=asc&since=2021-12-20T00%3A00%3A00%2B00%3A00>; rel=\"last\", <https://api.github.com/repositories/1296269/stargazers?per_page=100&page=1&sort=updated&direction=asc&since=2021-12-20T00%3A00%3A00%2B00%3A00>; rel=\"first\"",
    "X-RateLimit-Limit": "5000",
    "X-RateLimit-Remaining": "921",
    "X-RateLimit-Reset": "1640015555",
    "X-RateLimit-Used": "4079",
    "X-RateLimit-Resource": "core",
    "Access-Control-Expose-Headers": "ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Used, X-RateLimit-Resource, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type, X-GitHub-SSO, X-GitHub-Request-Id, Deprecation, Sunset",
    "Access-Control-Allow-Origin": "*",
    "Strict-Transport-Security": "max-age=31536000; includeSubdomains; preload",
    "X-Frame-Options": "deny",
    "X-Content-Type-Options": "nosniff",
    "X-XSS-Protection": "0",
    "Referrer-Policy": "origin-when-cross-origin, strict-origin-when-cross-origin",
    "Content-Security-Policy": "default-src 'none'",
    "Content-Encoding": "gzip",
    "X-GitHub-Request-Id": "DE0C:CC89:1760597:1832969:61C09D2F"
  },
  "raw": {
    "decode_content": false,
    "reason": "OK",
    "status": 200,
    "version": 11
  },
  "reason": "OK",
  "request": {
    "body": "PH%2y",
    "headers": {
      "User-Agent": "python-requests/2.26.0",
      "Accept-Encoding": "gzip, deflate",
      "Accept": "application/vnd.github.v3.star+json",
      "Connection": "keep-alive"
    },
    "method": "GET",
    "url": "https://api.github.com/repos/octocat/Hello-World/stargazers?per_page=100&page=13&sort=updated&direction=asc&since=2021-12-20T00%3A00%3A00%2B00%3A00"
  },
  "status_code": 200,
  "url": "https://api.github.com/repos/octocat/Hello-World/stargazers?per_page=100&page=13&sort=updated&direction=asc&since=2021-12-20T00%3A00%3A00%2B00%3A00"
}
```